### DISCUSSION TITLE
CubicChunks support

### DISCUSSION THREAD

**Discussion by yungnickyoung** [yungnickyoung/Better-Caves#22] [Suggestions]

**Comment by Digitalworldkamikaze** [comment]
How long it'll take? Should I wait for it or this is some sort of "future goal" that has a chance to never be done? Adding such thing would be really cool I mean.

**Comment by yungnickyoung** [comment]
Moreso the latter. It would be cool indeed but it's quite the task

**Comment by Digitalworldkamikaze** [comment]
> Moreso the latter. It would be cool indeed but it's quite the task

How can I help so? I don't know java well, but I'm capable of reading 100 manual pages per night

**Comment by yungnickyoung** [comment]
Well.. it's hard to say. A few of the changes that would need to be made:
- Most of the code would need to be refactored to use CC's "cubes" instead of Minecraft chunks
- Noise generation and interpolation would need to be changed accordingly, so instead of calculating noise for an entire column in a chunk, it would probably need to limit columns to being 16 blocks tall (one "cube")
- Surface detection would need to be rewritten to avoid unnecessary long checks for caves that are far underground
- The cavern design would need to be fundamentally changed. Currently, caverns only generate near bedrock (by default).. but this obviously can't be the case with CC. Maybe generate caverns in layers every 100 blocks?

Those are just a few things off the top of my head
